types/jest/index.d.ts

```diff
@@ -693,67 +693,56 @@ declare namespace jest {
         getState(): MatcherState & Record<string, any>;
     }
 
-    type JestMatchers<T> = JestMatchersShape<Matchers<void, T>, Matchers<Promise<void>, T>>;
+    type PromiseOf<T> = T extends Promise<infer S> ? S : never;
```

Comment:
+1 for this - can you add a test that resolve gets the unwrapped promise

Reply:
`
function promiseFn():Promise<{thingy:boolean}>{
    return Promise.resolve({thingy:true});
}
expect(promiseFn()).resolves.toEqual({thingy:true});
// $ExpectError
expect(promiseFn()).resolves.toEqual({notthingy:true});
`